src/Lykke.Job.NeoGasDistributor/Workflow/CommandHandlers/ExecuteDistributionPlanCommandHandler.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Threading.Tasks;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.Common.Log;
+using Lykke.Cqrs;
+using Lykke.Job.NeoGasDistributor.Domain.Services;
+using NeoGasDistributor.Contract.Commands;
+
+namespace Lykke.Job.NeoGasDistributor.Workflow.CommandHandlers
+{
+    public class ExecuteDistributionPlanCommandHandler
+    {
+        private readonly IDistributionPlanService _distributionPlanService;
+        private readonly ILog _log;
+
+        public ExecuteDistributionPlanCommandHandler(
+            IDistributionPlanService distributionPlanService,
+            ILogFactory logFactory)
+        {
+            _distributionPlanService = distributionPlanService;
+            _log = logFactory.CreateLog(this);
+        }
+
+        [UsedImplicitly]
+        public async Task<CommandHandlingResult> Handle(
+            ExecuteDistributionPlanCommand command,
+            IEventPublisher publisher)
+        {
+            if (await _distributionPlanService.PlanExistsAsync(command.PlanId))
+            {
+                try
+                {
+                    await _distributionPlanService.ExecutePlanAsync(command.PlanId);
+                }
+                catch (Exception)
+                {
+                    var retryDelay = TimeSpan.FromSeconds(30);
+                
+                    return CommandHandlingResult.Fail(retryDelay);
+                }
+            }
+            else
+            {
+                _log.Warning
```

Comment:
Looks like constency error - we should not omit this

Reply:
Nope, it will not be a problem: each payment in plan has unique id to protect from duplication in a ME level

Reply:
Which conditions leads to this case?

Reply:
Atm this command can be sent only manually via HTTP endpoint, where plan existance is checked too. But for the further modifications better to throw/retry in case if plan is not found.